src/wallet/wallet_constructor.rs

```diff
@@ -0,0 +1,226 @@
+use super::*;
+
+#[derive(Clone)]
+pub(crate) struct WalletConstructor {
+  ord_client: reqwest::blocking::Client,
+  name: String,
+  no_sync: bool,
+  rpc_url: Url,
+  settings: Settings,
+}
+
+impl WalletConstructor {
+  pub(crate) fn new(name: String, no_sync: bool, settings: Settings, rpc_url: Url) -> Result<Self> {
+    let mut headers = HeaderMap::new();
+    headers.insert(
+      header::ACCEPT,
+      header::HeaderValue::from_static("application/json"),
+    );
+
+    if let Some((username, password)) = settings.credentials() {
```

Comment:
Let's just double check if we're doing this in two places, if so, we can add a function to settings that returns the headers.